### PR TITLE
(chore): Migrate CI to GitHub Actions

### PR DESCRIPTION
**Pull request by rashil2000**
Relates to https://github.com/ScoopInstaller/Main/pull/3361

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rashil2000** [comment]
Pinging @r15ch13 to disable the AppVeyor pipeline

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          path: 'my_bucket'
+      - name: Checkout Scoop
+        uses: actions/checkout@v2
+        with:
+          repository: ScoopInstaller/Scoop
+          path: 'scoop_core'
+      - name: Init and Test
+        shell: powershell
+        run: |
+          $env:SCOOP_HOME="$(Resolve-Path '.\scoop_core')"
```
Comment (niheaven) [inline .github/workflows/ci.yml:26]:
```suggestion
          $env:SCOOP_HOME="$(Convert-Path '.\scoop_core')"
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          path: 'my_bucket'
+      - name: Checkout Scoop
+        uses: actions/checkout@v2
+        with:
+          repository: ScoopInstaller/Scoop
+          path: 'scoop_core'
+      - name: Init and Test
+        shell: pwsh
+        run: |
+          $env:SCOOP_HOME="$(Resolve-Path '.\scoop_core')"
```
Comment (niheaven) [inline .github/workflows/ci.yml:46]:
```suggestion
          $env:SCOOP_HOME="$(Convert-Path '.\scoop_core')"
```

File: bin/test.ps1 [hunk 3]
```diff
@@ -1,2 +1,5 @@
+#Requires -Modules @{ ModuleName = 'Pester'; MaximumVersion = '4.99' }
+
 if(!$env:SCOOP_HOME) { $env:SCOOP_HOME = Resolve-Path (scoop prefix scoop) }
```
Comment (niheaven) [inline bin/test.ps1:3]:
```suggestion
if (!$env:SCOOP_HOME) { $env:SCOOP_HOME = Convert-Path (scoop prefix scoop) }
```

File: bin/test.ps1 [hunk 4]
```diff
@@ -1,2 +1,5 @@
...
-Invoke-Pester "$psscriptroot/.."
+$result = Invoke-Pester "$psscriptroot/.." -PassThru
```
Comment (niheaven) [inline bin/test.ps1:4]:
```suggestion
$result = Invoke-Pester "$PSScriptRoot\.." -PassThru
```

**Review by niheaven — CHANGES_REQUESTED** [review]

**Review by niheaven — APPROVED** [review]

**Comment by Jimimaku** [comment]
> Relates to [ScoopInstaller/Main#3361](https://github.com/ScoopInstaller/Main/pull/3361)
> 
> * [x]  I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

**Comment by Jimimaku** [comment]
> Relates to [ScoopInstaller/Main#3361](https://github.com/ScoopInstaller/Main/pull/3361)
> 
> * [x]  I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

**Pull request merged**